pyaedt/generic/DataHandlers.py

```diff
@@ -482,6 +486,35 @@ def to_aedt(code):
     return return_code
 
 
+def str_to_bool(s):
+    """Convert a ``"True"`` or ``"False"`` string to its corresponding Boolean value.
+
+    If the passed arguments are not relevant in the context of conversion, the argument
+    itself is returned.  This method can be called using the `map()` function to
+    ensure conversion of Boolean strings in a list.
+
+    Parameters
+    ----------
+    s: str
+         Input parameter will be converted to Bool if the type is ``str`` with a value ``"True``, ``"true"``,
+         ``"False"`` or ``"false"``.
+
+    Returns
+    -------
+    bool
+
+    """
+    if type(s) == str:
+        if s.lower() == "true":
+            return True
+        elif s.lower() == "false":
+            return False
+        else:
+            return s
+    else:
+        return s
```

Comment:
@dcrawforAtAnsys I would suggest to raise an exception if `s` is not a `str`. Because if it is an `int` for instance it will return an `int` and not a `str` as the name of the method suggest. It could be confusing for the user.

Reply:
@MaxJPRey the intention is to pass any argument through the function unchanged if it is not a `str`.  This is used with `map()` to quickly parse the getter method in the native API.  See for example `Hfss.set_differntial_pair()`.